wasm-glow-ballers-lib/src/lib.rs

```diff
@@ -5,13 +5,110 @@ use std::collections::HashMap;
 use error::Error;
 use walrus::Module;
 
-pub fn mapping(wasm: &[u8]) -> Result<HashMap<String, usize>, Error> {
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone)]
+pub enum MappingItemType {
+    Global,
+    Function,
+    Memory,
+    Table,
+}
+
+// // Creates a mapping of all exported and imported globals (original use case of the library)
+// pub fn mapping(wasm: &[u8]) -> Result<HashMap<String, usize>, Error> {
+//     mapping_with_type(wasm, MappingItemType::Global)
+// }
+
+// Creates a mapping of all exported and imported items of the specified type
+pub fn mapping_with_type(
+    wasm: &[u8],
+    item_type: &MappingItemType,
+) -> Result<HashMap<String, usize>, Error> {
+    let module = Module::from_buffer(wasm).map_err(|_| Error::ParsingFailed)?;
+    let mut result = HashMap::new();
+
+    // Add exports
+    let exports = mapping_exports_with_type_and_module(&module, &item_type)?;
```

Comment:
When running `cargo clippy` in a terminal on your current commit, you will get the following warning:
`this expression creates a reference which is immediately dereferenced by the compiler`
pointing to the expression `&item_type`, which is dereferenced directly by the callee (as its signature doesn't take a reference anymore thanks to commit 7de6ab13269db9c52d552b0764ef9a4ad28fcee2).
You can fix it & check for all `clippy` warnings that point out the same suggestion.